integr8ly/tutorial-web-app-operator
# INTLY-2791: Add new OPENSHIFT_OAUTH_HOST param

**JIRA:** https://issues.jboss.org/browse/INTLY-2791
This PR is tied to https://github.com/integr8ly/tutorial-web-app/pull/495 change.

**Verification steps:**
Install this operator into a new NS:
```
oc new-project INTLY-2791-verification
make cluster/prepare
sed -i -e 's/v0.0.24/INTLY-2791/g' deploy/operator.yaml
make cluster/deploy
```
Create WebApp custom resource with appropriate values for your cluster:
```
apiVersion: integreatly.org/v1alpha1
kind: WebApp
metadata:
  name: solution-explorer
  namespace: INTLY-2791-verification
spec:
  app_label: tutorial-web-app
  template:
    parameters:
      OPENSHIFT_HOST: console-openshift-console.apps.omatskiv.cloudservices.rhmw.io
      OPENSHIFT_OAUTH_HOST: oauth-openshift.apps.omatskiv.cloudservices.rhmw.io
      OPENSHIFT_OAUTHCLIENT_ID: integreatly-solution-explorer
      SSO_ROUTE: placeholder
    path: /home/tutorial-web-app-operator/deploy/template/tutorial-web-app.yml
```
Create OAuthClient (with appropriate redirect url host):
```
kind: OAuthClient
apiVersion: oauth.openshift.io/v1
metadata:
  name: integreatly-solution-explorer
secret: test
redirectURIs:
  - >-
    https://tutorial-web-app-integreatly-solution-explorer.apps.omatskiv.cloudservices.rhmw.io
grantMethod: auto
```
Open tutorial-web-app route in private window.
You should be redirected to OpenShifts authorization page.
Once you login, you should be redirected back to tutorial web app.
You will likely just see "Provisioning services for your new environment." with no progress. If you check browser console, you will notice CORS related error messages. This is expected at this moment.

**Note:** Full E2E verification is currently blocked by inability to set additional CORS domain in OpenShift 4.1. This PR will be updated once we have more info on this issue.
I will do version bump and manifest PR once this PR is reviewed.
